packages/runtime-core/src/index.ts

```diff
@@ -48,7 +48,7 @@ export { defineAsyncComponent } from './apiAsyncComponent'
 
 // For getting a hold of the internal instance in setup() - useful for advanced
 // plugins
-export { getCurrentInstance } from './component'
+export { getCurrentInstance, PublicAPIComponent } from './component'
```

Comment:
I wonder at this point, if it still makes sense to have two exported types for a Component (with `Component` type) and if it's not worth joining them both under `Component` o rename `PublicAPIComponent` into something else.

I remember the `{ new ...} ` thing was some kind of a hack to get typings but I don't know more

Reply:
I would prefer if `defineComponent` was just returning the options (has it's implementation does)

```ts
export function defineComponent<T extends ComponentPropsOptions>(options: ComponentPropsOption): ComponentPropsOptions;
//etc
```

This way the typing would be much simple on `defineComponent` and the type and implementation would be the same.

Altho this would shift the "type resolving" to `h` and `component` but possibly it could be resolved with a type helper.